### ISSUE TITLE
[Feature Request]: Remap caps lock key

### ISSUE THREAD

**Issue by green7ea** [pop-os/cosmic-epoch#991]
I usually remap the caps lock key to be a control key to relieve stress on the little finger. This seems like a small feature, but my muscle memory makes using cosmic jarring – other than this, it's a joy to use.

**Feature**: caps lock can be mapped to another key (escape, control, ...). 

I would be willing to work on this but would need some guidance:

- Is this something that you're interested in?
- Should it be included in the base settings or be an external app?
- What should the option look like?
- Where should I start reading the code (some Rust knowledge, no cosmic knowledge)?

This is possible under other systems

## Gnome Tweaks

![image](https://github.com/user-attachments/assets/e755a2f6-d2fe-4c6b-a8e2-7a66883c2164)

## MacOS

![image](https://github.com/user-attachments/assets/f902c6c0-c369-4af0-a8e0-942178d721ba)

**Comment by git-f0x** [comment]
It is possible to do it with a config file (add `options: Some("caps:ctrl_modifier"),` to `~/.config/cosmic/com.system76.CosmicComp/v1/xkb_config`), but not sure when/if it will be added to Settings.

**Comment by green7ea** [comment]
This works very well, thank you so much. Would it be worth trying to turn this into a pull request? The data structure is already in place it seems.

**Comment by git-f0x** [comment]
Not sure if there are designs for that yet, or how/where that would be exposed in the UI.
You could possibly join the COSMIC Epoch channel on the Pop!_OS Mattermost for discussion, if you're interested in implementing it.

**Comment by green7ea** [comment]
It looks like the ability to remap the caps lock key was recently added but it didn't have the option to remap to control. [Here is a pull request](https://github.com/pop-os/cosmic-settings/pull/695) for that small change.

I was able to add and test cosmic settings but unfortunately not how it behaves in the compositor (I'm still running alpha-2).

I noticed that there doesn't seem to be any translations for this menu. I would be able to add some for French and German if that's a plus.
